utils/powertop/Makefile

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2019 <lucian.cristian@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=powertop
+PKG_VERSION:=2.10
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-v$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://01.org/sites/default/files/downloads/
+PKG_HASH:=d3b7459eaba7d01c8841dd33a3b4d369416c01e9bd8951b0d88234cf18fe4a75
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-v$(PKG_VERSION)
+PKG_MAINTAINER:=Lucian Cristain <lucian.cristian@gmail.com>
+PKG_LICENSE:=GPL-2.0
+
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+
+define Package/powertop
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS:=+libpci +libstdcpp +libncursesw +libnl-genl +zlib
```

Comment:
does uclibc++ work here?

Reply:
```
/home/build/proxy/staging_dir/target-x86_64_musl/usr/include/uClibc++/algorithm: In instantiation of 'void std::sort(RandomAccessIterator, RandomAccessIterator, Compare) [with RandomAccessIterator = devuser**; Compare = bool (*)(devuser*, devuser*)]':
devlist.cpp:329:51:   required from here
/home/build/proxy/staging_dir/target-x86_64_musl/usr/include/uClibc++/algorithm:836:14: error: 'stable_sort' was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
   stable_sort(first, last, comp);
   ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
```
and 


```
main.cpp: In function 'void one_measurement(int, int, char*)':
main.cpp:226:3: error: 'pthread_t' was not declared in this scope
   pthread_t thread = 0UL;
   ^~~~~~~~~
main.cpp:226:3: note: suggested alternative: 'pread'
   pthread_t thread = 0UL;
   ^~~~~~~~~
   pread
main.cpp:228:23: error: 'thread' was not declared in this scope
   if (pthread_create(&thread, NULL, measure_background_thread, &sample_interval))
                       ^~~~~~
main.cpp:228:23: note: suggested alternative: 'pread'
   if (pthread_create(&thread, NULL, measure_background_thread, &sample_interval))
                       ^~~~~~
                       pread
main.cpp:228:7: error: 'pthread_create' was not declared in this scope
   if (pthread_create(&thread, NULL, measure_background_thread, &sample_interval))
       ^~~~~~~~~~~~~~
main.cpp:228:7: note: suggested alternative: 'timer_create'
   if (pthread_create(&thread, NULL, measure_background_thread, &sample_interval))
       ^~~~~~~~~~~~~~
       timer_create
main.cpp:234:7: error: 'thread' was not declared in this scope
   if (thread)
       ^~~~~~
main.cpp:234:7: note: suggested alternative: 'pread'
   if (thread)
       ^~~~~~
       pread
main.cpp:237:4: error: 'pthread_join' was not declared in this scope
    pthread_join( thread, NULL);
    ^~~~~~~~~~~~
main.cpp:237:4: note: suggested alternative: 'thread_local'
    pthread_join( thread, NULL);
    ^~~~~~~~~~~~
    thread_local

```

Reply:
First issue is one that's fixed by my uclibc++ patch on OpenWrt's patchwork that everyone refuses to touch.

Second issue is a missing header.

Reply:
Just noticed as I'm building that this is a build requirement: https://github.com/openwrt/packages/pull/8228

Reply:
it was stated in the first message

Reply:
This builds fine locally. Looks like I need to submit a dumber version of the patch.

Reply:
Ah false alarm. It fails later on as long long unsigned int is not supported with the >> operator. Although I thought I patched that locally... Will need to investigate.

For right now, uclibc++ is not possible.